backend/coreapp/models/github.py

```diff
@@ -65,13 +66,31 @@ def details(self) -> NamedUser:
         if cached:
             return cached
 
-        details = Github(self.access_token).get_user_by_id(self.github_id)
+        try:
+            details = Github(self.access_token).get_user_by_id(self.github_id)
+        except BadCredentialsException:
+            # Uh oh - the user we're getting details for has an invalid token
+            # Revoke their session to force them to log in again ...
+            all_sessions = Session.objects.filter(expire_date__gte=now())
+
+            user_sessions = [
+                i.pk
+                for i in all_sessions
+                if i.get_decoded().get("_auth_user_id") == str(self.user.id)
+            ]
+
+            Session.objects.filter(pk__in=user_sessions).delete()
+
+            # ... and fallback to using the github api unauthenticated
+            # This *does* have a much stricter rate limit of 60 requests per minute,
+            # which is why we tried using an auth token first
+            details = Github().get_user_by_id(self.github_id)
 
         cache.set(cache_key, details, API_CACHE_TIMEOUT)
         return details
 
     def __str__(self) -> str:
-        return "@" + self.details().login
+        return "@" + self.user.username
```

Comment:
Happy to change this over to user ID if I'm missing something, but: I think this should preserve the behavior we have now, but without the network request?

Reply:
seems good to me!